template.go

```diff
@@ -174,24 +147,33 @@ func (s *Set) getTemplate(templatePath string) (t *Template, err error) {
 		}
 	}
 
-	return s.getTemplateFromLoader(templatePath)
+	t, err = s.getTemplateFromLoader(templatePath)
+	if err == nil && !s.developmentMode {
+		s.templates[templatePath] = t
```

Comment:
Access to `s.templates` should either be guarded with a mutex or use the type [`sync.Map`](https://golang.org/pkg/sync/#Map), as I suspect this type is likely to be a high read on single write.

Reply:
will be addressed in fix for #181